compiler/kernel/op.go

```diff
@@ -101,7 +101,7 @@ func (b *Builder) BuildVamToSeqFilter(filter dag.Expr, d demand.Demand, poolID,
 		return nil, err
 	}
 	cache := b.source.Lake().VectorCache()
-	search, err := vop.NewSearcher(b.rctx, cache, l, pool, e, demand.Fields(d))
+	search, err := vop.NewSearcher(b.rctx, cache, l, pool, e, fields)
```

Comment:
Nit: Move the FieldsOf call into this method. 
```suggestion
	project, _ := optimzer.FieldsOf(filter)
	search, err := vop.NewSearcher(b.rctx, cache, l, pool, e, project)
```